homeassistant/components/zwave/__init__.py

```diff
@@ -908,7 +908,11 @@ def _check_entity_ready(self):
         if polling_intensity:
             self.primary.enable_poll(polling_intensity)
 
-        platform = get_platform(self._hass, component, DOMAIN)
+        integration = asyncio.run_coroutine_threadsafe(
```

Comment:
That is not good, we probably need provide `loader.get_integration()`

Reply:
I may have bit off more than I can chew currently, there's a lot of `get_platform` mocking in tests that I've never worked with, so it may be better for someone else to take this on

Reply:
Nah, we should just replace this code with:

```python
        platform = importlib.import_module('.{}'.format(conf[CONF_PLATFORM]),
                                           __name__)
```

Reply:
Because integration code shouldn't load an integration object to load it's own code.

Reply:
And then everywhere in the tests that mock get_platform, mock `importlib.import_module`